akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -703,6 +703,33 @@ final case class Referer(uri: Uri) extends jm.headers.Referer with RequestHeader
   def getUri: akka.http.javadsl.model.Uri = uri.asJava
 }
 
+//https://tools.ietf.org/html/rfc7231#section-7.1.3
+sealed trait `Retry-After` extends jm.headers.RetryAfter
+final case class RetryAfterDuration(delay: Long) extends `Retry-After` with ResponseHeader {
```

Comment:
Our current pattern for these cases is a bit different. We don't make the header class an ADT (sorry if that was unclear), but give it a parameter with the type of the ADT. So it would be more like

```scala
final case class `Retry-After`(retryAfter: RetryAfter) extends jm.headers.RetryAfter with ResponseHeader { ... }
```

and then in an extra file

```scala
sealed abstract class RetryAfter
final case class RetryAfterDuration(delay: Long) extends RetryAfter
...
```

It's a bit more verbose but it follows our best practices.

Reply:
that was what I understood in the beginning, but then something made me change my mind, it's ok, I'll fix that soon, thanks

Reply:
just to know: why `sealed abstract class` and not `sealed trait` for the hierarchy root?

and also... can I make the ADT leaves as _Value Classes_?